File: src/main/java/com/alealogic/singleproxy/repository/CustomerRepository.java
package com.alealogic.singleproxy.repository;

import com.alealogic.singleproxy.entity.Customer;
import org.springframework.data.jpa.repository.JpaRepository;

public interface CustomerRepository extends JpaRepository<Customer, Long> {

    Customer findCustomerByApiKey(String apiKey);
}

File: src/main/java/com/alealogic/singleproxy/repository/CustomerRepository.kt
package com.alealogic.singleproxy.repository

import com.alealogic.singleproxy.entity.Customer
import org.springframework.data.jpa.repository.JpaRepository

interface CustomerRepository : JpaRepository<Customer?, Long?> {
    fun findCustomerByApiKey(apiKey: String?): Customer?
}